### ISSUE TITLE
`key-spacing` ignores parens

### ISSUE THREAD

**Issue by seahindeniz** [eslint-stylistic/eslint-stylistic#189]
### Describe the bug

Following examples throws error:
```
interface Foo {
    bar: 123;
    baz: (123);
}
```

But these don't:
```ts
interface Fab {
    bar:(
        123
    );
    baz:(
        | 123
        | 321
        | 222
    );
}
```

The reason why I use parens in `Fab.baz` is to avoid getting `indent` errors that expects the following
```ts
interface Fab {
    bar:(
        123
    );
    baz:
    | 123
    | 321
    | 222;
}
```

### Reproduction

https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-nbjlp9?file=playground%2Feslint.config.js,playground%2Findex.ts

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
Thanks for bringing this up. We consider this as a niche bug that only occurs in specific cases or with specific setups/configurations. In general, we agree this should be fixed, but maintainers may or may not have enough incentive to work on it.

We are counting on the original poster or any volunteer from the community to send a Pull Request to fix this.
